## Data Analyse for our data

In [7]:
# External Imports
import numpy as np
import pandas as pd

# Internal Imports

# Read data
data = pd.read_csv('./private/files/heart.csv')

# Analyze datas
data.describe().round(2)

age      sex       cp  trestbps     chol      fbs  restecg  \
count  1025.00  1025.00  1025.00   1025.00  1025.00  1025.00  1025.00   
mean     54.43     0.70     0.94    131.61   246.00     0.15     0.53   
std       9.07     0.46     1.03     17.52    51.59     0.36     0.53   
min      29.00     0.00     0.00     94.00   126.00     0.00     0.00   
25%      48.00     0.00     0.00    120.00   211.00     0.00     0.00   
50%      56.00     1.00     1.00    130.00   240.00     0.00     1.00   
75%      61.00     1.00     2.00    140.00   275.00     0.00     1.00   
max      77.00     1.00     3.00    200.00   564.00     1.00     2.00   

       thalach    exang  oldpeak    slope       ca     thal   target  
count  1025.00  1025.00  1025.00  1025.00  1025.00  1025.00  1025.00  
mean    149.11     0.34     1.07     1.39     0.75     2.32     0.51  
std      23.01     0.47     1.18     0.62     1.03     0.62     0.50  
min      71.00     0.00     0.00     0.00     0.00     0.00     0.00  
25%     132.00     0.00     0.00     1.00     0.00     2.00     0.00  
50%     152.00     0.00     0.80     1.00     0.00     2.00     1.00  
75%     166.00     1.00     1.80     2.00     1.00     3.00     1.00  
max     202.00     1.00     6.20     2.00     4.00     3.00     1.00

# 1- 
La distribution de l'âge des individus dans le jeu de données montre une répartition variée, allant de 29 à 77 ans, avec des fréquences variables dans chaque tranche d'âge.

In [8]:
# Look for correlation between age and cardiovascular disease
proportions = data.groupby('sex')['target'].mean()
# Affichage des proportions
print("Proportions de maladie cardiaque par sexe :")
print(proportions)

Proportions de maladie cardiaque par sexe :
sex
0    0.724359
1    0.420757
Name: target, dtype: float64


# 2-
Le résultat indique les proportions de maladie cardiaque pour chaque sexe.

- Pour le sexe 0 (supposément féminin, puisque 0 peut être généralement utilisé pour représenter le féminin dans de nombreux jeux de données), la proportion de maladie cardiaque est d'environ 0.724 ou 72.4%.
- Pour le sexe 1 (supposément masculin, puisque 1 peut être généralement utilisé pour représenter le masculin), la proportion de maladie cardiaque est d'environ 0.421 ou 42.1%.

Cela indique qu'il y a une différence significative dans la présence de maladie cardiaque entre les sexes dans votre jeu de données. En particulier, les femmes (sexe 0) semblent avoir une proportion plus élevée de maladie cardiaque par rapport aux hommes (sexe 1). Cette différence pourrait être explorée davantage pour comprendre les facteurs sous-jacents et les implications cliniques.

In [9]:
# Calculer les proportions de maladie cardiaque pour chaque type de douleur thoracique
proportions = data.groupby('cp')['target'].mean()

# Afficher les résultats
print("Proportions de maladie cardiaque pour chaque type de douleur thoracique :")
print(proportions)

Proportions de maladie cardiaque pour chaque type de douleur thoracique :
cp
0    0.245473
1    0.802395
2    0.771127
3    0.662338
Name: target, dtype: float64


# 3-
Les résultats indiquent les proportions de maladie cardiaque pour chaque type de douleur thoracique (cp)  :

- Pour le type de douleur thoracique 0, la proportion de maladie cardiaque est d'environ 0.245 ou 24.5%.
- Pour le type de douleur thoracique 1, la proportion de maladie cardiaque est d'environ 0.802 ou 80.2%.
- Pour le type de douleur thoracique 2, la proportion de maladie cardiaque est d'environ 0.771 ou 77.1%.
- Pour le type de douleur thoracique 3, la proportion de maladie cardiaque est d'environ 0.662 ou 66.2%.

Ces résultats suggèrent qu'il existe une différence significative dans la proportion de maladie cardiaque entre les différents types de douleur thoracique. En particulier, les types de douleur thoracique 1, 2 et 3 semblent avoir des proportions plus élevées de maladie cardiaque par rapport au type de douleur thoracique 0. Cela suggère une relation potentielle entre le type de douleur thoracique et la présence de maladie cardiaque dans votre jeu de données.

In [10]:
# Calculer les valeurs moyennes pour chaque variable en fonction de la présence ou de l'absence de maladie cardiaque
mean_values = data.groupby('target')[['trestbps', 'thalach', 'chol']].mean()

# Afficher les résultats
print("Valeurs moyennes pour chaque variable en fonction de la présence ou de l'absence de maladie cardiaque :")
print(mean_values)

Valeurs moyennes pour chaque variable en fonction de la présence ou de l'absence de maladie cardiaque :
          trestbps     thalach        chol
target                                    
0       134.106212  139.130261  251.292585
1       129.245247  158.585551  240.979087


# 4-
Les valeurs moyennes pour chaque variable en fonction de la présence ou de l'absence de maladie cardiaque sont les suivantes :

- Pour les individus sans maladie cardiaque (target = 0) :
  - Pression artérielle au repos moyenne : environ 134.11 mmHg
  - Fréquence cardiaque maximale moyenne : environ 139.13 bpm
  - Cholestérol moyen : environ 251.29 mg/dL

- Pour les individus avec maladie cardiaque (target = 1) :
  - Pression artérielle au repos moyenne : environ 129.25 mmHg
  - Fréquence cardiaque maximale moyenne : environ 158.59 bpm
  - Cholestérol moyen : environ 240.98 mg/dL

Ces valeurs moyennes fournissent des informations sur les différences moyennes dans ces variables entre les individus avec et sans maladie cardiaque. Par exemple, les individus avec maladie cardiaque semblent avoir une pression artérielle au repos légèrement plus basse, une fréquence cardiaque maximale plus élevée et un cholestérol légèrement plus bas en moyenne par rapport aux individus sans maladie cardiaque.

In [11]:
# Calculer les proportions de maladie cardiaque pour les deux groupes de glycémie à jeun
proportions = data.groupby('fbs')['target'].mean()

# Afficher les résultats
print("Proportions de maladie cardiaque pour les individus avec fbs <= 120 mg/dl et fbs > 120 mg/dl :")
print(proportions)

Proportions de maladie cardiaque pour les individus avec fbs <= 120 mg/dl et fbs > 120 mg/dl :
fbs
0    0.521789
1    0.464052
Name: target, dtype: float64


# 5-
Les résultats montrent les proportions de maladie cardiaque pour les individus avec une glycémie à jeun inférieure ou égale à 120 mg/dl (fbs = 0) et pour ceux avec une glycémie à jeun au-dessus de 120 mg/dl (fbs = 1)  :

- Pour les individus avec une glycémie à jeun inférieure ou égale à 120 mg/dl, la proportion de maladie cardiaque est d'environ 0.522 ou 52.2%.
- Pour les individus avec une glycémie à jeun au-dessus de 120 mg/dl, la proportion de maladie cardiaque est d'environ 0.464 ou 46.4%.

Cela indique qu'il y a une légère différence dans la proportion de maladie cardiaque entre les deux groupes, mais elle n'est pas très significative. Dans cet ensemble de données, une glycémie à jeun plus élevée ne semble pas être fortement associée à une présence accrue de maladie cardiaque.

In [12]:
# Calculer les proportions de maladie cardiaque pour chaque valeur de la variable exang
proportions = data.groupby('exang')['target'].mean()

# Afficher les résultats
print("Proportions de maladie cardiaque pour chaque valeur de la variable exang :")
print(proportions)

Proportions de maladie cardiaque pour chaque valeur de la variable exang :
exang
0    0.669118
1    0.205797
Name: target, dtype: float64


# 6- 
Les résultats montrent les proportions de maladie cardiaque pour chaque valeur de la variable exang (angine induite par l'exercice)     :

- Pour les individus sans angine induite par l'exercice (exang = 0), la proportion de maladie cardiaque est d'environ 0.669 ou 66.9%.
- Pour les individus avec angine induite par l'exercice (exang = 1), la proportion de maladie cardiaque est d'environ 0.206 ou 20.6%.

Cela indique qu'il y a une corrélation significative entre l'angine induite par l'exercice et la présence de maladie cardiaque dans votre jeu de données. En particulier, les individus avec angine induite par l'exercice semblent avoir une proportion nettement plus faible de maladie cardiaque par rapport à ceux sans angine induite par l'exercice.

1. **Différences de maladie cardiaque entre les sexes** : Les données indiquent une différence significative dans la proportion de maladie cardiaque entre les sexes. Les femmes semblent avoir une proportion plus élevée de maladie cardiaque que les hommes dans cet ensemble de données.

2. **Relation entre le type de douleur thoracique et la maladie cardiaque** : Il existe une association notable entre le type de douleur thoracique et la présence de maladie cardiaque. Les individus avec certains types de douleur thoracique (notamment les types 1, 2 et 3) présentent des proportions plus élevées de maladie cardiaque que ceux avec le type 0.

3. **Impact des variables physiologiques sur la maladie cardiaque** : Les variables physiologiques telles que la pression artérielle au repos, la fréquence cardiaque maximale et le taux de cholestérol semblent être associées à la présence de maladie cardiaque. Par exemple, les individus atteints de maladie cardiaque ont tendance à avoir des niveaux plus bas de pression artérielle au repos et de cholestérol, mais des fréquences cardiaques maximales plus élevées que ceux sans maladie cardiaque.

4. **Relation entre l'angine induite par l'exercice et la maladie cardiaque** : Il existe une corrélation significative entre l'angine induite par l'exercice et la présence de maladie cardiaque. Les individus avec angine induite par l'exercice ont une proportion nettement plus faible de maladie cardiaque que ceux sans angine induite par l'exercice.

5. **Impact de la glycémie à jeun sur la maladie cardiaque** : Bien qu'il y ait une légère différence dans la proportion de maladie cardiaque entre les individus avec une glycémie à jeun inférieure ou égale à 120 mg/dl et ceux avec une glycémie à jeun supérieure à 120 mg/dl, cette différence n'est pas très significative dans cet ensemble de données.

Ces déductions fournissent un aperçu des facteurs qui pourraient être associés à la maladie cardiaque dans votre jeu de données, ce qui pourrait être utile pour orienter des analyses plus approfondies ou des décisions cliniques.